### PR TITLE
make things more pydantic

### PR DESCRIPTION
**Pull request by aparcar**

### CODE REVIEW

**Comment by efahl** [comment]
Let me know when you've got this all sorted.  My changes to the API stuff have been working towards understanding how the `Annotated` and `Field` and other things work with redoc, so I can make a bunch of additions and incorporate all the `description` data from the old `openapi.yml`.

Which reminds me, is there some way I can generate the docs/redoc pages locally so I can see how it all fits together?

**Comment by aparcar** [comment]
Please check the readme, it's as easy as poetry install and running the fastapi dev command

**Comment by efahl** [comment]
I was just finishing up https://github.com/openwrt/asu/pull/914 when you submitted this one...  Closely related.

File: tests/test_package_changes.py [hunk 1]
```diff
@@ -1,65 +1,76 @@
...
 def test_apply_package_changes_adds_kmod_switch_rtl8366s():
-    req = {
-        "version": "23.05",
-        "target": "ath79/generic",
-        "profile": "buffalo_wzr-hp-g300nh-s",
-        "packages": ["kmod-ath9k-htc"],
-    }
-    appy_package_changes(req)
+    build_request = BuildRequest(
+        **{
```
Comment (efahl) [inline tests/test_package_changes.py:7]: For these, you could also say
```
... = BuildRequest(
    version = "23.05",
    target = "ath79/generic",
    ...
)
```
Although the current form is more suggestive of the json that it represents, so...

**Pull request merged**